more_itertools/more.py

```diff
@@ -545,3 +550,62 @@ def _get_values(self, value):
 
     def __getitem__(self, value):
         return self._get_values(value)
+
+
+def interleave(*iterables):
+    """Return a new iterable yielding from each iterable in turn,
+    until the shortest is exhausted. Note that this is the same as
+    chain(*zip(*iterables)).
+
+    >>> list(interleave([1, 2, 3], [4, 5], [6, 7, 8]))
+    [1, 4, 6, 2, 5, 7]
+    """
+    return chain.from_iterable(zip(*iterables))
+
+
+def interleave_longest(*iterables):
+    """Return a new iterable yielding from each iterable in turn,
+    skipping any that are exhausted. Note that this is not the same as
+    chain(*zip_longest(*iterables)).
+
+    >>> list(interleave_longest([1, 2, 3], [4, 5], [6, 7, 8]))
+    [1, 4, 6, 2, 5, 7, 3, 8]
+    """
+    i = chain.from_iterable(zip_longest(*iterables, fillvalue=_marker))
+    return filter(lambda x: x is not _marker, i)
+
+
+def collapse(iterable, basetype=None, levels=None):
+    """Flatten an iterable containing some iterables (themselves containing
+    some iterables, etc.) into non-iterable types, strings, elements
+    matching ``isinstance(element, basetype)``, and elements that are
+    ``levels`` levels down.
+
+    >>> list(collapse([[1], 2, [[3], 4], [[[5]]], 'abc']))
+    [1, 2, 3, 4, 5, 'abc']
+    >>> list(collapse([[1], 2, [[3], 4], [[[5]]]], levels=2))
+    [1, 2, 3, 4, [5]]
+    >>> list(collapse((1, [2], (3, [4, (5,)])), list))
+    [1, [2], 3, [4, (5,)]]
+    """
+    def walk(node, level):
+        if (
```

Comment:
Ah, but this style is in PEP 8! See last examples in the [indentation section](https://www.python.org/dev/peps/pep-0008/#id17). I generally run my commits through [pycodestyle](https://github.com/PyCQA/pycodestyle) to check for PEP 8 violations.